Source/WebCore/Modules/mediastream/MediaDevices.cpp

```diff
@@ -351,7 +350,7 @@ void MediaDevices::exposeDevices(Vector<CaptureDeviceWithCapabilities>&& newDevi
             deviceId = center.hashStringWithSalt(newDevice.persistentId(), deviceIDHashSalts.ephemeralDeviceSalt);
         else
             deviceId = center.hashStringWithSalt(newDevice.persistentId(), deviceIDHashSalts.persistentDeviceSalt);
-        auto groupId = newDevice.groupId().isEmpty() ? emptyString() : hashedGroupId(newDevice.groupId());
+        auto groupId = center.hashStringWithSalt(newDevice.groupId(), deviceIDHashSalts.ephemeralDeviceSalt);
```

Comment:
Why do we always use the ephemeral salt now?

Reply:
groupId is hashed by a per document salt. The ephemeral salt (computed on UIProcess side) is per document so we keep that per-document property and we can reduce from 3 salts to just 2.